staging/src/k8s.io/pod-security-admission/admission/admission.go

```diff
@@ -141,7 +141,7 @@ func (DefaultPodSpecExtractor) PodSpecResources() []schema.GroupResource {
 	return retval
 }
 
-func extractPodSpecFromTemplate(template *corev1.PodTemplateSpec) (*metav1.ObjectMeta, *corev1.PodSpec, error) {
+func ExtractPodSpecFromTemplate_original(template *corev1.PodTemplateSpec) (*metav1.ObjectMeta, *corev1.PodSpec, error) {
```

Comment:
Questions:
1. have you considered wrapping the existing admission plugin with the additional modifications we need rather than this patch?
2. how likely a user might see an error coming from a mutated pod (ie. coming from SCC updates) that won't be explicitly obvious to him reading the DC, for example?

Reply:
> how likely a user might see an error coming from a mutated pod (ie. coming from SCC updates) that won't be explicitly obvious to him reading the DC, for example?

on balance, I think the false warning message is a worse outcome and one we're getting field complaints about.


> have you considered wrapping the existing admission plugin with the additional modifications we need rather than this patch?

Doing this would mean duplicating the logic for the pod spec identification and extraction and the injection point remains very similar patch-wise.  It also means duplicating config consumption logic since this current happens after that evaluation and we'd need to coordinate the two to wrap it.  Not impossible, but if I were maintaining it, I think I'd rather maintain this.